### PR TITLE
Create env

### PR DESCRIPTION
**Pull request by tomrsae**

### CODE REVIEW

File: snake_env.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        self._turn_snake(action)
+
+        ate = self.state.update()
+        won = self.state.has_won()
+        reward = 0
+        
+        observation = self._get_obs()
+        if ate:
+            reward = 1 / self.state.steps
+        if won:
```
Comment (chrisryda) [inline snake_env.py:55]: Should prob think about adding a reward for moving towards the apple

File: snake_env.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+        import pygame
+
+        if self.screen is None:
+            pygame.init()
+            pygame.display.init()
+            self.screen = pygame.display.set_mode((self.window_size, ) * 2)
+        if self.clock is None:
+            self.clock = pygame.time.Clock()
+
```
Comment (chrisryda) [inline snake_env.py:98]: Delete 1 newline

File: snake_state.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+from typing import Any, Tuple
+import random
+from enum import Enum
+
+DIRECTIONS = [
+    (0, 1), # Up
+    (1, 0),  # Right
+    (0,-1), # Down
+    (-1,0) # Left
+]
```
Comment (magsteen) [inline snake_state.py:12]: Directions defined 2 different places. should be one place

**Review by magsteen — CHANGES_REQUESTED** [review]

**Review by chrisryda — APPROVED** [review]
Great job, ready to merge!

**Review by ngorke — APPROVED** [review]
I like it! Picasso :art:  :paintbrush:

**Review by magsteen — APPROVED** [review]

**Pull request merged**